examples/threading/threading.c

```diff
@@ -0,0 +1,77 @@
+#include "threading.h"
+#include <unistd.h>
+#include <stdlib.h>
+#include <string.h>
+#include <stdio.h>
+#include <syslog.h>
+#include <errno.h>
+
+// Optional: use these functions to add debug or error prints to your application
+#define DEBUG_LOG(msg,...)
+//#define DEBUG_LOG(msg,...) printf("threading: " msg "\n" , ##__VA_ARGS__)
+#define ERROR_LOG(msg,...) printf("threading ERROR: " msg "\n" , ##__VA_ARGS__)
+
+void* threadfunc(void* thread_param)
+{
+    int ret = 0;
+
+    struct thread_data* thread_func_args = (struct thread_data *)thread_param;
+
+    openlog(NULL, 0, LOG_USER);
+
+    // Wait for the specified time before attempting to obtain the mutex
+    usleep(thread_func_args->wait_to_obtain_ms * 1000);
```

Comment:
@krishnasuhagiya29 missing a check for usleep status here.

Reply:
Yes, Thanks!